addons/zylann.hterrain/tools/brush/brush_editor_overlay.gd

```diff
@@ -0,0 +1,99 @@
+@tool
+extends Control
+
+const MIN_UI_CIRCLE_SIZE = 40
+const MAX_UI_CIRCLE_SIZE = 500
+
+@onready var _brush_size_background: TextureRect = %BrushSizeBackground
+@onready var _brush_size_preview: TextureRect = %BrushSizePreview
+@onready var _value_label: Label = %ValueLabel
+@onready var _overlay_name_label: Label = %OverlayNameLabel
+@onready var _exponential_slider: HSlider = %ExponentialSlider
+
+@export var brush_size_factor: float = 2.5
+@export var min_value: float = -1
+@export var max_value: float = -1
+var _brush_preview_color: Color = Color.LIGHT_GREEN
+var _dpi_scale: float = 1.0
+var _value: float = 0.0
+
+signal on_value_selected(new_value: int)
+signal on_cancel
+
+var background_margin: int = 10
+
+
+func _physics_process(delta: float) -> void:
+	_update_size(_get_mouse_distance())
+
+
+func _input(event: InputEvent) -> void:
+	if event is InputEventMouseButton:
+		if event.button_index == MOUSE_BUTTON_LEFT and event.pressed:
+			on_value_selected.emit(_value)
+		else:
+			on_cancel.emit()
+	if event is InputEventKey:
+		if event.keycode == KEY_ESCAPE and event.pressed:
+			on_cancel.emit()
+
+
+func set_brush_preview_color(brush_color):
+	_brush_preview_color = brush_color
+	_update_brush_preview_color()
+
+
+func _update_brush_preview_color():
+	_brush_size_preview.modulate = _brush_preview_color
+
+
+func set_overlay_name(overlay_label_name: String) -> void:
+	_overlay_name_label.text = overlay_label_name
+
+
+func _update_size(value: float) -> void:
+	var dist = clampi(value * brush_size_factor, MIN_UI_CIRCLE_SIZE*_dpi_scale, MAX_UI_CIRCLE_SIZE*_dpi_scale )
+	var ui_size := clampi(dist, MIN_UI_CIRCLE_SIZE*_dpi_scale, MAX_UI_CIRCLE_SIZE*_dpi_scale)
+	_brush_size_background.size = Vector2(ui_size + background_margin, ui_size + background_margin)
+	_brush_size_background.position = Vector2(-( (ui_size/2) + (background_margin/2)) , -( (ui_size/2) + (background_margin/2)))
+	_brush_size_preview.size = Vector2(ui_size, ui_size)
+	_brush_size_preview.position = Vector2(-(ui_size/2) , -(ui_size/2))
+
+	_exponential_slider.min_value = MIN_UI_CIRCLE_SIZE*_dpi_scale
+	_exponential_slider.max_value = MAX_UI_CIRCLE_SIZE*_dpi_scale
+	_exponential_slider.value = (_exponential_slider.min_value+_exponential_slider.max_value)-ui_size
```

Comment:
Does it actually make sense to use exponential for every property?
Also I would probably have ported over the few calculations instead of instancing an entire control, but I guess it works

Reply:
Not sure, but it seems to work fine right now. After more people test it, we can change it if there is a need.

I was trying to avoid copying code, and since the slider is in cpp it should be faster anyways.